githubkit/core.py

```diff
@@ -323,33 +324,64 @@ def _check(
             error_models = error_models or {}
             status_code = str(response.status_code)
 
-            if response.status_code == 403 and "retry-after" in response.headers:
-                raise RateLimitExceededError(
-                    timedelta(seconds=int(response.headers["retry-after"])),
-                    original_message=response.content,
-                )
-
+            error_model = error_models.get(
+                status_code,
+                error_models.get(
+                    f"{status_code[:-2]}XX", error_models.get("default", Any)
+                ),
+            )
+            resp = Response(response, error_model)
+
+            # handle rate limit exceeded
+            # https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api#exceeding-the-rate-limit
+            # https://github.com/octokit/plugin-throttling.js/blob/135a0f556752a6c4c0ed3b2798bb58e228cd179a/src/index.ts#L134-L179
+
+            # Secondary rate limits
+            if response.status_code in (403, 429):
+                try:
+                    error = resp.json()
+                except Exception:
+                    error = None
+
+                # error message indicates that you exceeded a secondary rate limit
+                if (
+                    error
+                    and "message" in error
+                    and "secondary rate" in error["message"]
+                ):
```

Comment:
I'm not sure that this check will always work.

The documentation says: `an error message that indicates that you exceeded a secondary rate limit.`
A message that does not contain the words "secondary rate" could also match that description.
For (made up) example:
"Too many concurrent requests. No more than 100 concurrent requests are allowed."

Not sure if this every happens at the moment (or future moments).
But when it would happen and there is a `retry-after` response header but no `x-ratelimit-remaining` header, this implementation will not recognize the (secondary) rate limit.
This can be solved by moving the code from line 352 to 357 out of this if-block.
(see comment with edit suggestion)